enterprise-suite/templates/prometheus-deployment.yaml

```diff
@@ -180,9 +185,15 @@ spec:
       volumes:
         - name: config-volume
           emptyDir: {}
+        # By default, console-api-defaults will be mounted twice. Template for prometheus config (prometheus.yml) will always
+        # be picked up from console-api-defaults-volume and default monitors (default-monitors.json, static-rules.yml) from
+        # console-api-volume. This allows to override default monitors volume while keeping default prometheus config template.
+        - name: console-api-defaults-volume
+          configMap:
+            name: console-api-defaults
```

Comment:
I think this is a bit too confusing. Instead can we just create a ConfigMap per console-api file, and let users override each? e.g. `console-api-static-rules`, `console-api-prometheus-config`, `console-api-default-monitors`. Then have a parameter for each that let's a user override them, `console-api { static-rules: console-api-static-rules }`, etc.

Alternatively we can try to be clever and merge a user ConfigMap with our "templates"/"bundled" config map. But that feels a bit too magic.

Reply:
Ok, but should we even allow overriding prometheus config? I think not. I'll expose separate chart values to override the other two.

Reply:
Probably not - but we might need to start making some parts of it configurable, like scrape time.